DB/src/main/java/io/deephaven/db/v2/locations/parquet/local/ParquetColumnLocation.java

```diff
@@ -0,0 +1,332 @@
+    }
+
+    @Override
+    public <TYPE> ColumnRegionObject<TYPE, Values> makeColumnRegionObject(@NotNull final ColumnDefinition<TYPE> columnDefinition) {
+        //noinspection unchecked
+        return new ParquetColumnRegionObject<>((ColumnChunkPageStore<Values>) getPageStore(columnDefinition));
+    }
+
+    @Override
+    public ColumnRegionInt<DictionaryKeys> makeDictionaryKeysRegion(@NotNull final ColumnDefinition<?> columnDefinition) {
+        // TODO (https://github.com/deephaven/deephaven-core/issues/857): Address multiple row groups (and thus offset adjustments for multiple dictionaries)
+        final ColumnChunkPageStore<DictionaryKeys> dictionaryKeysPageStore = getDictionaryKeysPageStore(columnDefinition);
+        return dictionaryKeysPageStore == null ? null : new ParquetColumnRegionInt<>(dictionaryKeysPageStore);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <TYPE> ColumnRegionObject<TYPE, Values> makeDictionaryRegion(@NotNull final ColumnDefinition<?> columnDefinition) {
+        // TODO (https://github.com/deephaven/deephaven-core/issues/857): Address multiple row groups (and thus multiple dictionary pages)
+        final Chunk<Values> dictionaryValuesChunk = (Chunk<Values>) getDictionary(columnDefinition);
+        return dictionaryValuesChunk == null ? null : ParquetColumnRegionSymbolTable.create(columnDefinition.getDataType(), dictionaryValuesChunk);
+    }
+
+    /**
+     * Get the {@link ColumnChunkPageStore} backing this column location.
+     *
+     * @param columnDefinition The {@link ColumnDefinition} used to lookup type information
+     * @return The page store
+     */
+    @NotNull
+    public final ColumnChunkPageStore<ATTR> getPageStore(@NotNull final ColumnDefinition<?> columnDefinition) {
+        fetchValues(columnDefinition);
+        return pageStore;
+    }
+
+    /**
+     * Get the {@link Dictionary} backing this column location.
+     *
+     * @param columnDefinition The {@link ColumnDefinition} used to lookup type information
+     * @return The dictionary, or null if it doesn't exist
+     */
+    @Nullable
+    public Chunk<ATTR> getDictionary(@NotNull final ColumnDefinition<?> columnDefinition) {
+        fetchValues(columnDefinition);
+        return dictionary;
+    }
+
+    /**
+     * Get the {@link ColumnChunkPageStore} backing the indices for this column location. Only usable when there's
+     * a dictionary.
+     *
+     * @param columnDefinition The {@link ColumnDefinition} used to lookup type information
+     * @return The page store
+     */
+    @Nullable
+    private ColumnChunkPageStore<DictionaryKeys> getDictionaryKeysPageStore(@NotNull final ColumnDefinition<?> columnDefinition) {
+        fetchValues(columnDefinition);
+        return dictionaryKeysPageStore;
+    }
+
+    private void fetchValues(@NotNull final ColumnDefinition<?> columnDefinition) {
+        try {
+            if (pageStoreCreator != null) {
```

Comment:
I get that the first if is inside the try block to make the double-check-locking idiom evident... but it does feel weird.
You could move the if outside the try and have both checks inverted (and the inner one just return).